scipy/special/tests/test_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from hypothesis import given, strategies
+import hypothesis.extra.numpy as npst
+from numpy.testing import assert_allclose
+
+from scipy.special._support_cupy_torch_jax import (get_array_special_func,
+                                                   array_special_func_map)
+from scipy.conftest import array_api_compatible
+from scipy import special
+from scipy._lib.array_api_compat.array_api_compat import numpy as np
+
+
+def test_dispatch_failure():
+    message = ("SciPy cannot dispatch to special function 'ndtr' "
+               "in array library 'object'.")
+    with pytest.raises(ValueError, match=message):
+        get_array_special_func('ndtr', xp=object)
+
+
+@array_api_compatible
+@given(data=strategies.data())
+@pytest.mark.parametrize('f_name_n_args', array_special_func_map.items())
+def test_cupy_torch_jax_support(xp, data, f_name_n_args):
+    f_name, n_args = f_name_n_args
+    f = getattr(special, f_name)
+    dtypes = ['float32', 'float64']  # good enough for me
```

Comment:
Is this enough for dtypes?

Reply:
That's [all the float dtypes required by the standard](https://data-apis.org/array-api/2022.12/API_specification/data_types.html), though I guess in principle you could also test integers.

Reply:
My guess was that we don't compile for integers because the output is always between zero and 1. I was thinking about complex, though.